packages/client/src/api/postMessages.ts

```diff
@@ -1,11 +1,30 @@
 import { api } from "../utils/generateApi";
 import type { ApiRequest, ApiResponse } from "../utils/generateApi";
 
-export const messages = api({
-  methodName: "msgs",
-  method: "post",
-  path: "/v2/fungible/msgs",
-});
+export const messages = async (request: MessagesRequest) => {
+  const messagesRequest = {
+    sourceAssetDenom: request.sourceAssetDenom,
+    sourceAssetChainId: request.sourceAssetChainId,
+    destAssetDenom: request.destAssetDenom,
+    destAssetChainId: request.destAssetChainId,
+    amountIn: request.amountIn,
+    amountOut: request.amountOut,
+    addressList: request.addressList,
+    operations: request.operations,
+    estimatedAmountOut: request.estimatedAmountOut,
+    slippageTolerancePercent: request.slippageTolerancePercent,
+    timeoutSeconds: request.timeoutSeconds,
+    postRouteHandler: request.postRouteHandler,
+    chainIdsToAffiliates: request.chainIdsToAffiliates,
+    enableGasWarnings: request.enableGasWarnings,
+  };
+
+  return api({
+    methodName: "msgs",
+    method: "post",
+    path: "/v2/fungible/msgs",
+  })(messagesRequest);
+};
```

Comment:
honestly this feels like regressing back to the old way of doing things, and I realize it's just one exception but it's still annoying

Reply:
I feel like the API should just ignore extra fields if they are passed

Reply:
or honestly just telling integrators they can't just pass the entire `route` response to `msgs` (cuz you can't)